tests/unit/conftest.py

```diff
@@ -177,13 +177,15 @@ def httpretty_():
 
 
 @pytest.fixture
-def application_instance(pyramid_request):
-    return factories.ApplicationInstance(
+def application_instance(db_session, pyramid_request):
+    ai = factories.ApplicationInstance(
         consumer_key=pyramid_request.lti_user.oauth_consumer_key,
         developer_key="TEST_DEVELOPER_KEY",
         provisioning=True,
         settings=ApplicationSettings({}),
     )
+    db_session.flush()
```

Comment:
Needed so ai has a non null id.

Reply:
Is this really needed? The tests seem to pass without it. I wonder if it might slow things down by adding an extra `flush()` to every test that uses this fixture

Reply:
Odd, I added it when the tests were setup differently and it was required. Not needed anymore :+1: